Lib/test/test_capi.py

```diff
@@ -999,6 +999,36 @@ async def foo(arg): return await arg  # Py 3.5
             self.assertEqual(ret, 0)
             self.assertEqual(pickle.load(f), {'a': '123x', 'b': '123'})
 
+    def test_py_config_isoloated_per_interpreter(self):
+        # A config change in one interpreter must not leak to out to others.
+        code = """if 1:
+        import sys, _testinternalcapi
+
+        # Any config value would do, this happens to be the one being
+        # double checked at the time this test was written.
+        config = _testinternalcapi.get_config()
+        config['int_max_str_digits'] = 55555
+        _testinternalcapi.set_config(config)
+        sub_value = _testinternalcapi.get_config()['int_max_str_digits']
+        assert sub_value == 55555, sub_value
+        """
+        before_config = _testinternalcapi.get_config()
+        assert before_config['int_max_str_digits'] != 55555
```

Comment:
For me, this test on int_max_str_digits belongs to test_int, but you can keep it here if you prefer.

Reply:
this one is really more of a test of how config behaves between interpreters to verify that they each get their own unique config. it doesn't have to use this int related config setting - any would do - it just happens to.  i'll add a comment to the code about that.